### PR TITLE
now results have status codes

### PR DESCRIPTION
**Pull request by thekemkid**
Should this be considered the solution to #46 (non 200's not recorded)?

### CODE REVIEW

**Comment by davidmarkclements** [comment]
Yes - but to verify we need tests for 1xx, 3xx, 4xx, 5xx responses

File: test/run.test.js [hunk 1]
```diff
@@ -179,3 +190,22 @@ test('run should callback with after a bailout', (t) => {
     t.end()
   }, 3000)
 })
+
+for (var i = 1; i <= 5; i++) closure(i)
```
Comment (davidmarkclements) [inline test/run.test.js]: we're node 4 and up, you can just use `let` instead of using func closure

File: test/helper.js [hunk 2]
```diff
@@ -50,6 +50,25 @@ function startHttpsServer () {
   return server
 }
 
+for (var i = 1; i <= 5; i++) closure(i)
+
```
Comment (davidmarkclements) [inline test/helper.js]: use `let`

File: test/run.test.js [hunk 3]
```diff
@@ -179,3 +190,22 @@ test('run should callback with after a bailout', (t) => {
...
+
+function closure (i) {
+  test('run should count all reply status codes', (t) => {
```
Comment (davidmarkclements) [inline test/run.test.js]: change name of test to reflect error code, i.e. 

 ` test(`run should count all ${i}xx status codes`, (t) => {`

File: lib/run.js [hunk 4]
```diff
@@ -150,9 +150,9 @@ function run (opts, cb) {
         finish: new Date(),
         connections: opts.connections,
         pipelining: opts.pipelining,
-        '2xx': statusCodes[1],
         'non2xx': statusCodes[0] + statusCodes[2] + statusCodes[3] + statusCodes[4]
```
Comment (davidmarkclements) [inline lib/run.js]: I'm not sure we should keep non2xx

**Comment by davidmarkclements** [comment]
couple of picks, but fine overall

I think while we're at it @thekemkid can we sort out #47 as it's semi-related

only 2xx should affect req per sec average and latency measurements

File: lib/run.js (hunk 4 above)
Comment (thekemkid) [inline lib/run.js]: It's used in tests, and its not very harmful to keep it in, I can remove if you prefer though

**Comment by thekemkid** [comment]
uhm, #47 is very related as it's this PR... 😝 

Should I modify the code to only calc latency for 2xx's?

File: lib/run.js (hunk 4 above)
Comment (mcollina) [inline lib/run.js]: it's used in the cli output (non-json)

**Comment by mcollina** [comment]
> Should I modify the code to only calc latency for 2xx's?

👍 Yes, please :)

**Comment by mcollina** [comment]
LGTM

File: lib/run.js (hunk 4 above)
Comment (davidmarkclements) [inline lib/run.js]: Should  cli output all status codes though?

Comment (mcollina) [inline lib/run.js]: Probably yes, the ones that it finds

**Pull request merged**